roles/k3s/master/tasks/main.yml

```diff
@@ -0,0 +1,17 @@
+---
+
+- name: Copy K3s service file
+  register: k3s_service
+  template:
+    src: "k3s.service.j2"
+    dest: "{{ systemd_dir }}/k3s.service"
+    owner: root
+    group: root
+    mode: 0755
+
+- name: Enable and check K3s service
```

Comment:
Let's rename to: "Ensure K3s service is enabled and started"
Remember Ansible is desired-state. All tasks should as much as possible be there to ensure that something is in the state desired, not to systematically change the state of things. That's why it's good to provide checksum to get_url, avoid using actions (restart, reload, start...) and rather use states (started, enabled...)

However, naming all tasks with "Ensure blabla..." ends up being heavy to the reader, but in your mind you should see it this way. For instance:
"Download K3s binary for 'arm64' architecture" is "Ensure K3s binary for 'arm64' architecture is downloaded"

Reply:
Thanks for clarifying the naming pattern for Ansible